apps/aecore/src/aec_tx_pool.erl

```diff
@@ -357,3 +371,21 @@ check_minimum_fee(Tx,_Hash) ->
         false -> {error, too_low_fee}
     end.
 
+check_minimum_gas_price(Tx,_Hash) ->
+    case aetx:lookup_gas_price(aetx_sign:tx(Tx)) of
+        none -> ok;
+        {value, GasPrice} -> int_check_minimum_gas_price(GasPrice)
+    end.
+
+-dialyzer({no_match, int_check_minimum_gas_price/1}).
+int_check_minimum_gas_price(GasPrice) ->
+    case GasPrice >= aec_governance:minimum_gas_price() of
+        true  -> ok;
+        false -> {error, too_low_gas_price}
+    end.
+
+int_gas_price(Tx) ->
+    case aetx:lookup_gas_price(Tx) of
+        none -> ?PSEUDO_GAS_PRICE;
+        {value, GP} when is_integer(GP), GP >= 0 -> GP
```

Comment:
This should only be called for proper Txs so this check is overly cautious?

Reply:
Correct.